janus/message-packaging.md

```diff
@@ -279,21 +279,26 @@ Internal message structure (for each recipient):
 }
 ```
 
-The Meta message's internal structure can be the same for each recipient, including using the same ephemeral keypair. However, each message is encrypted _for_ each recipient using X25519 just like any other message.
+The Meta message's internal structure can be the same for each recipient, including using the same ephemeral keypair. 
+However, each message is encrypted _for_ each recipient using X25519 just like any other message.
 
-The sender can then send a collection of Authcrypt Meta messages, one for each recipient, and one Authcrypt Elemental Message. The Cloud Agent can disperse the appropriate Authcrypt Meta message to each recipient, along with the Elemental Message.
+The sender can then send a collection of Authcrypt Meta messages, one for each recipient, and one Authcrypt Elemental Message. The Cloud Agent can 
+disperse the appropriate Authcrypt Meta message to each recipient, along with the Elemental Message.
 
-The recipients decrypt the Authcrypt Meta messages and verify it like any other Authcrypt message. Then it hashes Elemental and verifies it matches the hash in the Meta. Then it decrypts Elemental using the key in Meta.
+The recipients decrypt the Authcrypt Meta messages and verify it like any other Authcrypt message. Then it hashes Elemental and verifies it matches the hash `msg_hash` in the Meta.
+If the `msg_hash` does not match the hash of the Elemental then it knows that either the sender did not calculate the hash correctly or the Cloud agent tampered the data.   
```

Comment:
Adding a hash of the encrypted data seems like a homegrown approach to a MAC. Consider doing a MAC

Reply:
Resolved